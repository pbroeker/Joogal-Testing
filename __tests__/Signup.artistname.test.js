import React from 'react';
import Artistname from '../Signup/Signup.Artistname';
import { fireEvent, render } from '@testing-library/react-native';

describe('Signup Welcome', () => {
  const setSignup = jest.fn();
  const navigation = {navigate : jest.fn()};
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = render( <Artistname navigation={navigation} setSignup={setSignup} /> );
  })

  test('snapshot', () => {
    const { toJSON } =  renderedComponent;
    expect(toJSON()).toMatchSnapshot();
  })

  test('renders correctly', () => {
    const { getByText } = renderedComponent;
    expect(getByText('Do you have an artist name?')).toBeTruthy();
  })

  test('calls setSignup correctly', () => {
    const { getByText } = renderedComponent;
    fireEvent(getByText('âž¤'), 'onPress');
    expect(setSignup).toHaveBeenCalledTimes(1);
  })
})