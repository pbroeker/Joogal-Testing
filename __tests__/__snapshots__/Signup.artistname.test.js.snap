// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Welcome snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1d1e20",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": "20%",
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontFamily": "PoiretOne_400Regular",
          "fontSize": 30,
          "marginBottom": 30,
          "marginLeft": "10%",
          "width": "52%",
        }
      }
    >
      Do you have an artist name?
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": "45%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      onChangeText={[Function]}
      placeholder="Artist name"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "whitesmoke",
          "marginVertical": "1%",
          "padding": 14,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginTop": 100,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "teal",
          "borderRadius": 50,
          "marginRight": 30,
          "opacity": 1,
          "padding": 10,
          "width": 56,
        }
      }
      testID="artistContinueButton"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "600",
          }
        }
      >
        âž¤
      </Text>
    </View>
  </View>
</View>
`;
